src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,112 @@
+      let userTarget = channel?.better_channel_members.find(
+        (data) => data?.user?.id !== signedProfileId
+      );
+
+      if (type === 'ANONYMOUS') {
+        userTarget = channel?.better_channel_members.find(
+          (data) => data?.user?.id !== anonProfileId
+        );
+      }
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: userTarget?.user?.name || userTarget?.user?.username,
+        image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId; // change to use getChannelListInfo
+      channel.targetName = chatName?.name; // change to use getChannelListInfo
+      channel.targetImage = chatName?.image; // change to use getChannelListInfo
+
+      channel.channel = {...channel};
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members);
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
+  };
+
+  const sendMessageDM = async (
+    id,
+    source: 'post' | 'comment',
+    channelCategory: ChannelCategory
+  ) => {
+    const initChat =
+      channelCategory === 'ANONYMOUS'
+        ? await initChatFromPostAnon({source, id})
+        : await initChatFromPost({source, id});
+
+    const channel = {
+      ...initChat?.data?.channel,
+      better_channel_members: initChat?.data?.better_channel_members,
+      members: initChat?.data?.better_channel_members
+    };
+
+    if (!channel?.members?.length) {
+      console.info('no members');
+      return;
+    }
+
+    if (channel?.type === 'topics') {
+      console.info('type is topics');
+      return;
+    }
+
+    try {
+      await Promise.all(
+        (channel?.members || []).map(async (member) => {
+          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+            channel?.id,
+            uuid(),
+            member // change to use getChannelListInfo
+          );
+          await userMember.saveOrUpdateIfExists(localDb);
+          await memberSchema.save(localDb);
+        })
+      );
+
+      await Promise.all(
+        (channel?.messages || []).map(async (message) => {
+          if (message?.type === 'deleted') return;
+          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          await chat.save(localDb);
+        })
+      );
+
+      await saveChannelList(channel, channelCategory);
+    } catch (e) {
+      console.log('error on saveChannelData', e);
+    }
+  };
+
+  return {
+    sendMessageDM
+  };
+};
+
+export default useDMMessage;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useDMMessage` hook seems to be well implemented. It uses hooks for local database, user authentication and chat utilities. The error handling is also done properly in the `saveChannelList` and `sendMessageDM` functions. However, there are a few points that could be improved:

1. In line 49 and 103, instead of just logging the error, it would be better to handle it or throw it so that the calling function can decide what to do with it.
2. In lines 39-41, comments suggest changes to use `getChannelListInfo`. If this function provides a more efficient or cleaner way to get the information, it should be used.
3. In line 86, a comment suggests changes to use `getChannelListInfo`. If this function provides a more efficient or cleaner way to get the information, it should be used.

```diff
  const saveChannelList = async (channel, type: ChannelCategory) => {
    try {
      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
      let userTarget = channel?.better_channel_members.find(
        (data) => data?.user?.id !== signedProfileId
      );

      if (type === 'ANONYMOUS') {
        userTarget = channel?.better_channel_members.find(
          (data) => data?.user?.id !== anonProfileId
        );
      }

      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
      const chatName = {
        name: userTarget?.user?.name || userTarget?.user?.username,
        image: userTarget?.user?.image || DEFAULT_PROFILE_PIC_PATH
      };

      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
      channel.myUserId = getChannelListInfo(signedProfileId); // change to use getChannelListInfo
      channel.targetName = getChannelListInfo(chatName?.name); // change to use getChannelListInfo
      channel.targetImage = getChannelListInfo(chatName?.image); // change to use getChannelListInfo

      channel.channel = {...channel};
      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members);
      await channelList.saveIfLatest(localDb);
      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
      return goToChatScreen(channelList);
    } catch (error) {
      throw new Error('sendMessage error: ' + error);
    }
  };

  const sendMessageDM = async (
    id,
    source: 'post' | 'comment',
    channelCategory: ChannelCategory
  ) => {
    const initChat =
      channelCategory === 'ANONYMOUS'
        ? await initChatFromPostAnon({source, id})
        : await initChatFromPost({source, id});

    const channel = {
      ...initChat?.data?.channel,
      better_channel_members: initChat?.data?.better_channel_members,
      members: initChat?.data?.better_channel_members
    };

    if (!channel?.members?.length) {
      console.info('no members');
      return;
    }

    if (channel?.type === 'topics') {
      console.info('type is topics');
      return;
    }

    try {
      await Promise.all(
        (channel?.members || []).map(async (member) => {
          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
            channel?.id,
            uuid(),
            getChannelListInfo(member) // change to use getChannelListInfo
          );
          await userMember.saveOrUpdateIfExists(localDb);
          await memberSchema.save(localDb);
        })
      );

      await Promise.all(
        (channel?.messages || []).map(async (message) => {
          if (message?.type === 'deleted') return;
          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
          await chat.save(localDb);
        })
      );

      await saveChannelList(channel, channelCategory);
    } catch (e) {
      throw new Error('error on saveChannelData: ' + e);
    }
  };

  return {
    sendMessageDM
  };
};

export default useDMMessage;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->